### PR TITLE
Add row conversion code from spark-rapids-jni

### PR DESCRIPTION
**Pull request by ttnghia**
This temporarily moves the row conversion code from spark-rapids-jni into libcudf. It is necessary to have the row conversion code compiled in a static library to overcome a CCCL issue that triggers invalid memory access when calling to `thrust::in(ex)clusive_scan` (https://github.com/NVIDIA/spark-rapids-jni/issues/1567).

In the future, when we have CCCL updated to fix the issue (1567), we may need to move the code back into spark-rapids-jni.

### CODE REVIEW

**Review by hyperbolic2346 — APPROVED** [review]

File: cpp/include/cudf/row_conversion.hpp
Comment (vyasr) [inline cpp/include/cudf/row_conversion.hpp:1]: Looks like this is just copied/moved from java/src/main/native/src/row_conversion.hpp

File: cpp/tests/transform/row_conversion.cpp
Comment (vyasr) [inline cpp/tests/transform/row_conversion.cpp:1]: Copied from https://github.com/NVIDIA/spark-rapids-jni/blob/branch-24.02/src/main/cpp/tests/row_conversion.cpp

File: cpp/src/transform/row_conversion.cu
Comment (vyasr) [inline cpp/src/transform/row_conversion.cu:1]: The style diff here is because the code in `java/` was not formatted with clang-format.

**Review by vyasr — APPROVED** [review]

**Review by mythrocks — APPROVED** [review]
Let's update the description to mention that this is a workaround for the failure in `spark-rapids-jni`.

File: cpp/src/transform/row_conversion.cu
Comment (mythrocks) [reply cpp/src/transform/row_conversion.cu:1]: That accounts for the style diffs.  There were material differences in the `spark-rapids-jni` version of `row_conversion.cu` that's not the easiest to spot in this diff.  This remains a temp fix, of course, until the CCCL issue is resolved.

File: cpp/src/transform/row_conversion.cu [hunk 1]
```diff
@@ -61,37 +60,40 @@
...
 using cuda::aligned_size_t;
 #else
-template <std::size_t> using aligned_size_t = size_t; // Local stub for cuda::aligned_size_t.
-#endif // ASYNC_MEMCPY_SUPPORTED
+template <std::size_t>
+using aligned_size_t = size_t;  // Local stub for cuda::aligned_size_t.
+#endif  // ASYNC_MEMCPY_SUPPORTED
 
 namespace cudf {
-namespace jni {
```
Comment (mythrocks) [inline cpp/src/transform/row_conversion.cu:94]: I admire your thoroughness in this, @ttnghia.

**Comment by hyperbolic2346** [comment]
Seems there was a cmake file missed

**Pull request merged**